src/main/java/com/google/cloud/genomics/dataflow/pipelines/AnnotateVariants.java

```diff
@@ -0,0 +1,307 @@
+    return transcripts;
+  }
+
+  private String retrieveReferenceBases(Genomics genomics, String referenceSetId, String refName,
+      long from, long to) throws IOException {
+    // TODO: Refactor once the References API supports query by reference name.
+    String refID = FluentIterable
+        .from(Paginator.References.create(genomics).search(
+            new SearchReferencesRequest().setReferenceSetId(referenceSetId)))
+        .filter(Predicates.compose(Predicates.equalTo(refName),
+            new Function<Reference, String>() {
+              @Override
+              public String apply(Reference ref) {
+                return ref.getName();
+              }
+            }))
+        .transform(new Function<Reference, String>() {
+          @Override
+          public String apply(Reference ref) {
+            return ref.getId();
+          }
+        })
+        .first().orNull();
+    if (Strings.isNullOrEmpty(refID)) {
+      throw new IllegalArgumentException(
+          "reference set " + referenceSetId + " has no reference named " + refName);
+    }
+
+    StringBuilder b = new StringBuilder();
+    String pageToken = "";
+    while (true) {
+      // TODO: Support full request parameterization for Paginator.References.Bases.
+      ListBasesResponse response = genomics.references().bases()
+          .list(refID)
+          .setStart(from)
+          .setEnd(to)
+          .setPageToken(pageToken)
+          .execute();
+      b.append(response.getSequence());
+      pageToken = response.getNextPageToken();
+      if (Strings.isNullOrEmpty(pageToken)) {
+        break;
+      }
+    }
+    LOG.info("retrieved " + b.length() + " bases");
+    return b.toString();
+  }
+
+  public static void main(String[] args) throws Exception {
+    GenomicsDatasetOptions opts = PipelineOptionsFactory.fromArgs(args)
+        .withValidation().as(GenomicsDatasetOptions.class);
+    GenomicsFactory.OfflineAuth auth = GenomicsOptions.Methods.getGenomicsAuth(opts);
+    Genomics genomics = auth.getGenomics(auth.getDefaultFactory());
+
+    List<String> callSetIds = ImmutableList.copyOf(opts.getCallSetIds().split(","));
+    List<String> transcriptSetIds =
+        validateAnnotationSetsFlag(genomics, opts.getTranscriptSetIds(), "TRANSCRIPT");
+    List<String> variantAnnotationSetIds =
+        validateAnnotationSetsFlag(genomics, opts.getVariantAnnotationSetIds(), "VARIANT");
+    String refsetId = validateRefsetForAnnotationSets(genomics, transcriptSetIds);
+
+    Iterable<Contig> contigs = opts.isAllReferences()
+        ? Contig.getContigsInVariantSet(
+            genomics, opts.getDatasetId(), SexChromosomeFilter.INCLUDE_XY)
+        : Contig.parseContigsFromCommandLine(opts.getReferences());
+
+    Pipeline p = Pipeline.create(opts);
+    DataflowWorkarounds.registerGenomicsCoders(p);
+    DataflowWorkarounds.getPCollection(ImmutableList.copyOf(contigs), p, opts.getNumWorkers())
```

Comment:
Is this necessary? Why not convert the list of Contigs to a PCollection directly?